core/schema/host.go

```diff
@@ -53,6 +60,61 @@ func (s *hostSchema) Install() {
 			}
 			return core.NewDirectory(parent, blobDef.ToPB(), "/", parent.Platform, nil), nil
 		}).Doc("Retrieves a content-addressed blob."),
+
+		dagql.Func("builtinContainer", func(ctx context.Context, parent *core.Query, args struct {
+			Digest string `doc:"Digest of the image manifest"`
+		}) (*core.Container, error) {
```

Comment:
re: https://github.com/dagger/dagger/pull/6704#issuecomment-1960300898 - are you thinking of asserting `dagql.IsInternal` here to avoid people depending on it? Or using `dagql.IsInternal` to conditionally register the API somehow? (That'd be neat)

There's an interesting consideration either way: you'd want to prevent these hidden/internal APIs from ending up in an ID, since that ID would then only work if it was part of an internal call.

I think everything that matters might still work since we don't want this to end up in any published IDs anyway (it's an impl. detail of the APIs that actually _do_ end up in IDs). But it's a little spooky. Just fyi!

Reply:
>  Or using dagql.IsInternal to conditionally register the API somehow? (That'd be neat)

Ya that's what I was imagining. And maybe also grouping everything in this category under like `Query.internal` or something just to avoid hidden namespace conflicts between user APIs and APIs they don't even know about. 

> There's an interesting consideration either way: you'd want to prevent these hidden/internal APIs from ending up in an ID that the user cares about, since that ID would then only work if it was part of an internal call.

Right right, I guess you could deal with that by returning always returning a `dagql.Instance` from these internal APIs such that the returned instance is constructed purely from non-internal APIs. So like `builtinContainer` could use `Select` to create a `Container` from scratch using `withRootfs`, `withEnvVariable`, etc. etc. and return that instance. 

I don't know if we could *enforce* this automatically or if it'd just need to be a manually enforced rule. But yeah definitely a medium sized wrench in that whole idea...